api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1138,7 +1138,10 @@ private HttpResponse executeReq(Method method, String region, String bucketName,
           errorXml = scanner.next();
         }
 
-        errorResponse = new ErrorResponse(new StringReader(errorXml));
+        if ("application/xml".equals(response.headers().get("content-type"))) {
+          errorResponse = new ErrorResponse(new StringReader(errorXml));
+        }
```

Comment:
now you are swallowing non-xml response.  you would need to throw an exception about non-xml response received.

Reply:
Done.